fetch/mesh_data.go

```diff
@@ -56,19 +56,40 @@ func (f *Fetch) getHashes(ctx context.Context, hashes []types.Hash32, hint datas
 				continue
 			}
 			if !res.IsLocal {
-				if err := receiver(ctx, res.Data); err != nil {
+				err := receiver(ctx, res.Data)
+				if err != nil {
 					f.logger.WithContext(ctx).With().Warning("failed to handle data",
 						log.String("hint", string(hint)),
 						log.String("hash", hash.String()),
 						log.Err(err))
 					errs = append(errs, err)
 				}
+				f.doneProcessing(res, err)
```

Comment:
will this run if request exits on context?

Reply:
no longer apply